src/data-source/collection-browser-data-source.ts

```diff
@@ -246,6 +246,19 @@ export class CollectionBrowserDataSource
     this.requestHostUpdate();
   }
 
+  /**
+   * @inheritdoc
+   */
+  resetPages(): void {
+    if (Object.keys(this.pages).length !== this.host.maxPagesToManage) {
```

Comment:
Q: This will still reset even if we already have more pages than needed. Would `<` instead of `!==` still work?